docs/user-guide/syntheticdata.rst

```diff
@@ -14,5 +14,743 @@ There are a variety of ways to construct synthetic data generation pipelines, wi
+
+.. code-block:: python
+
+    from openai import OpenAI
+    from nemo_curator import OpenAIClient
+    from nemo_curator.synthetic import NemotronGenerator
+
+    openai_client = OpenAI(
+        base_url="https://integrate.api.nvidia.com/v1",
+        api_key="<insert API key>"
+    )
+    client = OpenAIClient(openai_client)
+    generator = NemotronGenerator(client)
+
+    n_macro_topics = 20
+    model = "mistralai/mixtral-8x7b-instruct-v0.1"
+    model_kwargs = {
+        "temperature": 0.2,
+        "top_p": 0.7,
+        "max_tokens": 1024,
+    }
+
+    responses = generator.generate_macro_topics(
+        n_macro_topics=n_macro_topics, model=model, model_kwargs=model_kwargs
+    )
+
+    print(responses[0])
+    # Output:
+    # 1. Climate Change and Sustainable Living
+    # 2. Space Exploration and the Universe
+    # ...
+
+This example should seem very similar to the ``OpenAIClient.query_model``.
+We specify the model we are using just like before, along with additional keyword arguments to control the model's generation.
+``generator.generate_macro_topics`` queries the LLM and asks it to generate a list of topics about the world.
+There is an additional ``prompt_template`` parameter that is defaulted to the one used in Nemotron-4 340B, but it can be changed if desired.
+``responses`` will be a list of responses. There will be only one response unless ``n > 1`` is specified in ``model_kwargs``.
+
+The output of the above snippet will be a string response that contains a list of topics.
+Many LLM responses in the Nemotron pipeline will contain a list.
+Therefore, ``NemotronGenerator`` provides a helper function that will attempt to convert an LLM response into a Python list of strings
+
+.. code-block:: python
+
+    responses = generator.generate_macro_topics(
+        n_macro_topics=n_macro_topics, model=model, model_kwargs=model_kwargs
+    )
+
+    topic_list = generator.convert_response_to_yaml_list(
+        responses[0], model=model, model_kwargs=model_kwargs
+    )
+    print(topic_list[0])
+    # Output:
+    # Climate Change and Sustainable Living
+
+This helper function prompts an LLM to convert the previous response into a yaml format, then attempts to parse the yaml format.
+If the parsing fails, it will throw a ``YamlConversionError``.
+``topic_list`` is not guaranteed to have a length of 20.
+In our end to end pipelines that you will see later, NeMo Curator will raise a ``YamlConversionError`` if there is a mismatch between desired length of list and the received length of list, but this function does not check for it.
+
+With these examples out of the way, let's look at exactly how to replicate the Nemotron-4 340B synthetic data generation pipeline in NeMo Curator.
+For a more in-depth explanation of each of the steps, please refer to the `Nemotron-4 340B Technical Report <https://arxiv.org/abs/2406.11704v1>`_.
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Open Q&A Prompt Generation
```

Comment:
It seems like there's a lot of repeated text/examples in the next few sections, which maybe we want to trim down a little? Up to you.

Reply:
Yeah it is a bit repetitive, but I think it's good reference in case someone just wants to read part of the docs.